src/NServiceBus.Core/Pipeline/Incoming/TransportReceiveContext.cs

```diff
@@ -31,9 +30,7 @@ public TransportReceiveContext(IncomingMessage receivedMessage, TransportTransac
         /// </summary>
         public void AbortReceiveOperation()
         {
-            cancellationTokenSource.Cancel();
+            throw new OperationCanceledException();
```

Comment:
@danielmarbach can you clarify since it was mentioned in the analysis proposal?

Reply:
The problem is that this was a weird design from the start. AbortReceiveOperation previously just switched the state of the token that was passed in from the transport to the context but continued to execute the rest of the pipeline just to then within the transport after successful completion to check whether the current receive operation was canceled and then NACK the message. 

The assumption during the analysis was that the transport deals with the pipeline being canceled by observing cancellation exceptions anyway so, therefore, aborting the pipeline is equivalent to just throw an `OperationCanceledException`. That being said it would be possible to preserve the existing implementation and let the transport cancel the per receive token source. The consequence of this would be though that the remainder of the pipeline would still be executed up to the point where something actually observes the token and then at that stage the visible side effect of having canceled earlier the token source would be that another operation throws at some other place in the pipeline or if the token is not observed again only again in the transport itself (like it does today). Happy to further explain this in a call if needed/desired

Reply:
I definitely need a call to understand that 😆

Reply:
FTR we've agreed to obsolete this method and that will happen in another PR